cmake/kokkos_enable_options.cmake

```
@@ -73,6 +73,7 @@ KOKKOS_ENABLE_OPTION(HIP_MULTIPLE_KERNEL_INSTANTIATIONS OFF "Whether multiple ke
 
 # This option will go away eventually, but allows fallback to old implementation when needed.
 KOKKOS_ENABLE_OPTION(DESUL_ATOMICS_EXTERNAL OFF "Whether to use an external desul installation")
+KOKKOS_ENABLE_OPTION(ATOMICS_BYPASS OFF "**NOT RECOMMENDED** Whether to disable atomics")
```

I find it hard to understand what "disabling atomics" is supposed to mean. Maybe
```suggestion
KOKKOS_ENABLE_OPTION(ATOMICS_BYPASS OFF "**NOT RECOMMENDED** Whether to make atomics operations no-ops (only allowed for Serial-only builds).")
```

---

They are not no-ops, they do perform the operation, just not atomically.   That said I agree "disable atomics" is not great.

---

Yes, maybe we just go with "bypass" matching the options' name.

---

I find repeating the exact same word as the option not useful.  Because either you understand what "atomics bypass" mean and you don't care; or you are confused and you read the description and great that is not helpful.

---

Updated with Christian's suggestion